uberenv.py

```diff
@@ -234,6 +256,18 @@ class SpackEnv(UberEnv):
     def __init__(self, opts, extra_opts):
         UberEnv.__init__(self,opts,extra_opts)
 
+        if opts["package_name"]:
+            self.pkg_name = opts["package_name"]
+        else:
+            self.pkg_name = self.project_opts["package_name"]
```

Comment:
Probably not for package_name, I’ll add it.

Reply:
Since there are several ways to access a parameter, would it make sense to have a function that first checks the input args, then checks ``project.json`` and either throws an error or supplies a default value if it cannot be found?

Reply:
Also, it seems like we should require consistency between the entries in the command line args and in ``project.json``.